### ISSUE TITLE
Any package gives `Installation issue: ==> Error: Spec version is not concrete: mkfontdir `

### ISSUE THREAD

**Issue by punkpony** [LLNL/spack#26925]
### Steps to reproduce the issue

install of spack is as per documentation:

```
git clone -c feature.manyFiles=true https://github.com/spack/spack.git 
 . spack/share/spack/setup-env.sh 
more ~/.bash_profile 
```

Then I run installation of any software (tried gromacs and grace)
```
spack install grace 
```

I get the error:
```
==> Error: Spec version is not concrete: mkfontdir
```

I guess something is not right with the original instal of spack? But not sure how to check it.
Thank you!
V

### Information on your system

```
$ spack debug report 
* **Spack:** 0.16.3-5000-de8e795
* **Python:** 3.6.8
* **Platform:** linux-rhel7-power9le
* **Concretizer:** clingo
```

### Additional information

I don't seem to have build logs, m.b. as the install failed before? 

This is a general issue, not package related

### General information

- [X] I have run `spack debug report` and reported the version of Spack/Python/Platform
- [ ] I have run `spack maintainers <name-of-the-package>` and **@mentioned** any maintainers
- [ ] I have uploaded the build log and environment files
- [X] I have searched the issues of this repo and believe this is not a duplicate

**Comment by haampie** [comment]
can you provide the stacktrace with `spack -d <cmd>`

**Comment by punkpony** [comment]
- [check spec ] 

```
$ spack spec -I mkfontdir 
Input spec
--------------------------------
 -   mkfontdir

Concretized
--------------------------------
==> Error: Spec version is not concrete: mkfontdir
```

**Comment by punkpony** [comment]
Thank you @haampie 

Sorry got confused  with packages:

> can you provide the stacktrace with spack -d <cmd>

```
$ spack -d install grace
==> [2021-10-25-12:28:30.614685] Imported install from built-in commands
==> [2021-10-25-12:28:30.618516] Reading config file /projects/bddur15/spack/etc/spack/defaults/config.yaml
==> [2021-10-25-12:28:30.674236] Imported install from built-in commands
==> [2021-10-25-12:28:30.680394] Reading config file /projects/bddur15/spack/etc/spack/defaults/bootstrap.yaml
==> [2021-10-25-12:28:30.706662] Reading config file /projects/bddur15/spack/etc/spack/defaults/config.yaml
==> [2021-10-25-12:28:30.760206] DATABASE LOCK TIMEOUT: 3s
==> [2021-10-25-12:28:30.760270] PACKAGE LOCK TIMEOUT: No timeout
==> [2021-10-25-12:28:30.760868] Reading config file /projects/bddur15/spack/etc/spack/defaults/bootstrap.yaml
==> [2021-10-25-12:28:30.781996] DATABASE LOCK TIMEOUT: 3s
==> [2021-10-25-12:28:30.782059] PACKAGE LOCK TIMEOUT: No timeout
==> [2021-10-25-12:28:30.782153] [BOOTSTRAP CONFIG SCOPE] name=_builtin
==> [2021-10-25-12:28:30.782854] [BOOTSTRAP CONFIG SCOPE] name=defaults, path=/projects/bddur15/spack/etc/spack/defaults
==> [2021-10-25-12:28:30.782914] [BOOTSTRAP CONFIG SCOPE] name=defaults/linux, path=/projects/bddur15/spack/etc/spack/defaults/linux
==> [2021-10-25-12:28:30.782980] [BOOTSTRAP CONFIG SCOPE] name=bootstrap, path=/users/auntyv/.spack/bootstrap/config
==> [2021-10-25-12:28:30.783034] [BOOTSTRAP CONFIG SCOPE] name=bootstrap/linux, path=/users/auntyv/.spack/bootstrap/config/linux
==> [2021-10-25-12:28:30.783418] Reading config file /users/auntyv/.spack/bootstrap/config/linux/compilers.yaml
==> [2021-10-25-12:28:30.905917] [BOOTSTRAP ROOT SPEC] clingo: clingo-bootstrap@spack+python %gcc target=ppc64le
==> [2021-10-25-12:28:30.905986] [BOOTSTRAP MODULE clingo] Try importing from Python
==> [2021-10-25-12:28:30.906347] Reading config file /projects/bddur15/spack/etc/spack/defaults/bootstrap.yaml
==> [2021-10-25-12:28:31.137637] Reading config file /projects/bddur15/spack/etc/spack/defaults/config.yaml
==> [2021-10-25-12:28:31.339447] [BOOTSTRAP MODULE clingo] The installed spec "clingo-bootstrap@spack+python %gcc target=ppc64le ^python@3.6 /rfu4la457mdbpoffk2g5hikj2hhoek4s" provides the "clingo" Python module
==> [2021-10-25-12:28:31.339738] Reading config file /projects/bddur15/spack/etc/spack/defaults/repos.yaml
==> [2021-10-25-12:28:34.018272] Reading config file /etc/spack/compilers.yaml
==> [2021-10-25-12:28:34.076070] Reading config file /projects/bddur15/spack/etc/spack/defaults/config.yaml
==> [2021-10-25-12:28:34.131175] Reading config file /projects/bddur15/spack/etc/spack/defaults/packages.yaml
==> [2021-10-25-12:28:34.213133] Reading config file /etc/spack/packages.yaml
==> [2021-10-25-12:28:34.810352] SpackError: Spec version is not concrete: mkfontdir
==> [2021-10-25-12:28:34.810429] SpackError: Spec version is not concrete: mkfontdir
==> [2021-10-25-12:28:34.810488] Error: Spec version is not concrete: mkfontdir
Traceback (most recent call last):
  File "/projects/bddur15/spack/lib/spack/spack/cmd/__init__.py", line 164, in parse_specs
    spec.concretize(tests=tests)  # implies normalize
  File "/projects/bddur15/spack/lib/spack/spack/spec.py", line 2640, in concretize
    self._new_concretize(tests)
  File "/projects/bddur15/spack/lib/spack/spack/spec.py", line 2610, in _new_concretize
    result = spack.solver.asp.solve([self], tests=tests)
  File "/projects/bddur15/spack/lib/spack/spack/solver/asp.py", line 1691, in solve
    return driver.solve(setup, specs, dump, models, timers, stats, tests)
  File "/projects/bddur15/spack/lib/spack/spack/solver/asp.py", line 344, in solve
    solver_setup.setup(self, specs, tests=tests)
  File "/projects/bddur15/spack/lib/spack/spack/solver/asp.py", line 1399, in setup
    self.external_packages()
  File "/projects/bddur15/spack/lib/spack/spack/solver/asp.py", line 798, in external_packages
    for external_id, x in enumerate(external_specs)
  File "/projects/bddur15/spack/lib/spack/spack/solver/asp.py", line 798, in <listcomp>
    for external_id, x in enumerate(external_specs)
  File "/projects/bddur15/spack/lib/spack/spack/spec.py", line 3581, in version
    "Spec version is not concrete: " + str(self))
spack.error.SpecError: Spec version is not concrete: mkfontdir

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/projects/bddur15/spack/lib/spack/spack/main.py", line 816, in main
    return _invoke_command(command, parser, args, unknown)
  File "/projects/bddur15/spack/lib/spack/spack/main.py", line 529, in _invoke_command
    return_val = command(parser, args)
  File "/projects/bddur15/spack/lib/spack/spack/cmd/install.py", line 395, in install
    args.spec, concretize=True, tests=tests)
  File "/projects/bddur15/spack/lib/spack/spack/cmd/__init__.py", line 181, in parse_specs
    raise spack.error.SpackError(msg)
spack.error.SpackError: Spec version is not concrete: mkfontdir
```

**Comment by haampie** [comment]
right, so it seems there is something misconfigured in packages.yaml

**Comment by punkpony** [comment]
> right, so it seems there is something misconfigured in packages.yaml

What can I do about it?

**Comment by punkpony** [comment]
> right, so it seems there is something misconfigured in packages.yaml

I am not sure I am happy that there are different paths here:
```
==> [2021-10-25-12:28:34.131175] Reading config file /projects/bddur15/spack/etc/spack/defaults/packages.yaml
==> [2021-10-25-12:28:34.213133] Reading config file /etc/spack/packages.yaml
```

The cluster has an older spack version, which _may_ have a problem. It actually did give me the same error, hence my reinstall for the latest version of spack. I wonder if `/etc/spack` links there? 

So yep, the /etc/spack is from Nov 2020, i.e. olderr spack install. I think I need to get rid of it somewhere in my paths

**Comment by haampie** [comment]
Oof... that's spack's worst nightmare. I have an open PR to fix that: #26735, please upvote it so we can get it in before 0.17.

For now please just cherry-pick that PR and set SPACK_SYSTEM_CONFIG_PATH=/somewhere/else, for instance in your bash_profile or so.

Edit: fixed the PR number, there are many, 26735 is the correct one.

**Comment by punkpony** [comment]
So, yep, literally this, I have compilers.yaml and packages.yaml being picked up from the old install.

My root is correct:
```
$ echo $SPACK_ROOT 
/projects/bddur15/spack
```
and I cannot see the path to `/etc/spack` lingering anywhere ...

Will do the `SPACK_SYSTEM_CONFIG_PATH` thank you!

**Comment by punkpony** [comment]
Hi, @haampie & @tgamblin,  sorry, did not have time to check if setting `SPACK_SYSTEM_CONFIG_PATH` will solve this

 I have done:
`export SPACK_SYSTEM_CONFIG_PATH=$SPACK_ROOT/etc/spack/defaults`
`export SPACK_DISABLE_LOCAL_CONFIG=true`

unfortunately, I am still getting the following:

`$ spack -d install grace`
_skipping some rows that are fine
```
==> [2021-10-27-12:09:18.870118] Reading config file /projects/bddur15/spack/etc/spack/defaults/repos.yaml
==> [2021-10-27-12:09:21.602281] Reading config file /etc/spack/compilers.yaml
==> [2021-10-27-12:09:21.660483] Reading config file /projects/bddur15/spack/etc/spack/defaults/config.yaml
==> [2021-10-27-12:09:21.715346] Reading config file /projects/bddur15/spack/etc/spack/defaults/packages.yaml
==> [2021-10-27-12:09:21.797445] Reading config file /etc/spack/packages.yaml
==> [2021-10-27-12:09:22.409311] SpackError: Spec version is not concrete: mkfontdir
==> [2021-10-27-12:09:22.409389] SpackError: Spec version is not concrete: mkfontdir
==> [2021-10-27-12:09:22.409447] Error: Spec version is not concrete: mkfontdir
Traceback (most recent call last):
```

i.e. it is **STILL** looking into the original spack within the system at `/etc/spack`

Have I missed something somewhere?

**Comment by haampie** [comment]
I guess you are on an older commit than ae6e83b1d5ef2e7049fbc01a7f308b4b5bdf4c4b?

This issue got closed automatically when #26735 was merged

**Comment by punkpony** [comment]
Fair point 

> I guess you are on an older commit than [ae6e83b](https://github.com/spack/spack/commit/ae6e83b1d5ef2e7049fbc01a7f308b4b5bdf4c4b)?
oops, fair point... updated, seems good. Thank you!

```
==> [2021-10-27-13:18:55.200801] Reading config file /projects/bddur15/spack/etc/spack/defaults/bootstrap.yaml
==> [2021-10-27-13:18:55.222214] DATABASE LOCK TIMEOUT: 3s
==> [2021-10-27-13:18:55.222278] PACKAGE LOCK TIMEOUT: No timeout
==> [2021-10-27-13:18:55.222373] [BOOTSTRAP CONFIG SCOPE] name=_builtin
==> [2021-10-27-13:18:55.223070] [BOOTSTRAP CONFIG SCOPE] name=defaults, path=/projects/bddur15/spack/etc/spack/defaults
==> [2021-10-27-13:18:55.223129] [BOOTSTRAP CONFIG SCOPE] name=defaults/linux, path=/projects/bddur15/spack/etc/spack/defaults/linux
==> [2021-10-27-13:18:55.223193] [BOOTSTRAP CONFIG SCOPE] name=bootstrap, path=/users/auntyv/.spack/bootstrap/config
==> [2021-10-27-13:18:55.223246] [BOOTSTRAP CONFIG SCOPE] name=bootstrap/linux, path=/users/auntyv/.spack/bootstrap/config/linux
==> [2021-10-27-13:18:55.224164] Reading config file /users/auntyv/.spack/bootstrap/config/linux/compilers.yaml
==> [2021-10-27-13:18:55.332490] [BOOTSTRAP ROOT SPEC] clingo: clingo-bootstrap@spack+python %gcc target=ppc64le
==> [2021-10-27-13:18:55.332559] [BOOTSTRAP MODULE clingo] Try importing from Python
==> [2021-10-27-13:18:55.332923] Reading config file /projects/bddur15/spack/etc/spack/defaults/bootstrap.yaml
==> [2021-10-27-13:18:56.247408] Reading config file /projects/bddur15/spack/etc/spack/defaults/config.yaml
```

> This issue got closed automatically when #26735 was merged
yep, did not want to write in the merge.
